media-video/gaupol/gaupol-1.14.1-r1.ebuild

```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit distutils-r1 virtualx xdg-utils
+
+DESCRIPTION="A subtitle editor for text-based subtitles"
+HOMEPAGE="https://otsaloma.io/gaupol/ https://github.com/otsaloma/gaupol/"
+SRC_URI="https://github.com/otsaloma/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+IUSE="player-external +player-integrated spell"
+
+RDEPEND="
+	app-text/iso-codes
+	dev-python/chardet[${PYTHON_USEDEP}]
+	dev-python/pygobject:3[${PYTHON_USEDEP}]
+	x11-libs/gtk+:3[introspection]
+	player-external? (
+		|| (
+			media-video/mplayer
+			media-video/mpv
+			media-video/vlc
+		)
+	)
+	player-integrated? (
+		>=media-libs/gstreamer-1.6
+		media-libs/gst-plugins-bad
+		media-libs/gst-plugins-base
+		media-libs/gst-plugins-good
+		media-plugins/gst-plugins-gtk
+		media-plugins/gst-plugins-libav
+	)
```

Comment:
In general, this PR violates the policy guide: https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001

Please fix it to NOT use a USE flag, but instead e.g. use the optfeature.eclass functionality.

Alternatively, explain (in the commit message, at minimum) what makes these USE flags special enough to be a special case that breaks the general rule. :)

Reply:
Looking through the source code its overzealous while misses the pango use flag (checked documentation, textoverlay and timeoverlay are a part of the pango plugin).

media-libs/gstreamer
media-libs/gst-plugins-base[pango]
media-libs/gst-plugins-good

Discourages media-plugins/gst-plugins-vaapi

```
        print("GStreamer-vaapi found, known to sometimes cause trouble.",
              "If video doesn't play, please try uninstalling it.",
              "See https://github.com/otsaloma/gaupol/issues/79",
              file=sys.stderr)
```

https://github.com/otsaloma/gaupol/blob/1.14.1/gaupol/util.py#L157

Reply:
https://gitweb.gentoo.org/repo/gentoo.git/tree/app-i18n/translate-shell/translate-shell-0.9.7.1.ebuild#n34

Take inspiration from this.

Reply:
Alright, so it uses the with gobject-introspection and gst-python just gives it a nicer interface. So its correct its not needed.

One thing is that I'm now sure on the actual dependencies for the internal player.

optfeature "media-libs/gstreamer[introspection] media-libs/gst-plugins-base[pango] media-plugins/gst-plugins-gtk"

Gentoo splits the plugins differently so gst-plugins-good and the rest arent used at all.

Reply:
@parona-source thanks for the tip, I'm going to mention something like this in the ebuild:
```sh
# To check which plugins are required (vaapisink is mentioned in the code but it is not used):
plugins=$(qlist -e gaupol | xargs grep -ho 'Gst.ElementFactory.*' | cut -d'"' -f2 | sort -u | grep -vFx vaapisink)
xargs -n1 gst-inspect-1.0 <<< "$plugins" | awk '/Filename/ {print $2}' | sort -u | xargs qfile
```